lib/improver/tests/ensemble_calibration/ensemble_calibration/test_EnsembleCalibration.py

```diff
@@ -186,210 +178,335 @@ def test_basic_wind_speed_realizations(self):
         """
         Test that the plugin returns an iris.cube.CubeList
         of wind_speed cubes with the expected data, where the plugin
         returns a cubelist of, firstly, the predictor and, secondly the
-        variance.
-        The ensemble realizations is the predictor.
+        variance. The ensemble realizations is the predictor.
         """
-        import imp
-        try:
-            imp.find_module('statsmodels')
-            statsmodels_found = True
-            import statsmodels.api as sm
-            self.sm = sm
-        except ImportError:
-            statsmodels_found = False
-        if statsmodels_found:
-            predictor_data = np.array(
-                [[3.15758874, 10.63961216, 18.12163557],
-                 [25.60365899, 33.08568241, 40.56770583],
-                 [48.04972924, 55.53175266, 63.01377608]])
-            variance_data = np.array(
-                [[0.01406566, 0.01406566, 0.01406566],
-                 [0.01406566, 0.01406566, 0.01406566],
-                 [0.01406566, 0.01406566, 0.01406566]])
-        else:
-            predictor_data = np.array(
-                [[2.05799912, 9.73470204, 17.41140496],
-                 [25.08810788, 32.7648108, 40.44151372],
-                 [48.11821664, 55.79491955, 63.47162247]])
-            variance_data = np.array(
-                [[4.26987243, 4.26987243, 4.26987243],
-                 [4.26987243, 4.26987243, 4.26987243],
-                 [4.26987243, 4.26987243, 4.26987243]])
-        calibration_method = "ensemble model output_statistics"
         distribution = "truncated gaussian"
-        desired_units = "m s^-1"
         predictor_of_mean_flag = "realizations"
         plugin = Plugin(
-            calibration_method, distribution, desired_units,
+            self.calibration_method, distribution,
             predictor_of_mean_flag=predictor_of_mean_flag)
         calibrated_predictor, calibrated_variance = plugin.process(
             self.current_wind_speed_forecast_cube,
             self.historic_wind_speed_forecast_cube,
             self.wind_speed_truth_cube)
-        self.assertArrayAlmostEqual(calibrated_predictor.data, predictor_data,
-                                    decimal=4)
-        self.assertArrayAlmostEqual(calibrated_variance.data, variance_data)
+        self.assertArrayAlmostEqualLowerPrecision(
+            calibrated_predictor.data,
+            self.expected_specific_truncated_gaussian_predictor_data)
+        self.assertArrayAlmostEqualLowerPrecision(
+            calibrated_variance.data,
+            self.expected_specific_truncated_gaussian_variance_data)
+        self.assertArrayAlmostEqual(
+            calibrated_predictor.data,
+            self.expected_truncated_gaussian_mean_data, decimal=0)
+        self.assertArrayAlmostEqual(
+            calibrated_variance.data,
+            self.expected_truncated_gaussian_variance_data, decimal=0)
```

Comment:
This is intentional in this case, because these particular results have been calculated when the ensemble mean is used as a predictor, rather than the ensemble realizations. Note that for these tests, I've compared both against the specific results obtained using the ensemble realizations as the input but also against the results obtained using the ensemble mean as the input. This is just to show that the results are similar, but as the inputs to the ensemble calibration are different it could not be expected that these results would match exactly.

Reply:
I've added a comment in for each of the unit tests to explain that the assertion is intended to show that there is broad agreement (to 0 decimal places) that the results match between using the ensemble mean as the predictor and the ensemble realizations as the predictor.